app/uk/gov/hmrc/helptosavefrontend/controllers/SessionBehaviour.scala

```diff
@@ -41,12 +41,24 @@ trait SessionBehaviour {
       transformer: NINOLogMessageTransformer): Future[Result] =
     sessionCacheConnector.get
       .semiflatMap(_.fold(noSession)(whenSession))
-      .leftMap{
+      .leftMap {
         e ⇒
           logger.warn(s"Could not read sessions data from keystore: $e", htsContext.nino)
           internalServerError()
       }.merge
 
+  def updateSessionWithBackLink(session: HTSSession)(redirectAfterUpdate: ⇒ Result)(implicit htsContext: HtsContextWithNINO,
```

Comment:
this method isn't really updating the session with a back link - in general, it updates the session with whatever you give it as an input. The function isn't really redirecting after update either since `Result` could be doing something else. This function is really probably only `updateSession`. You may or may not need this function if you could just do:
```scala
    sessionCacheConnector.put(session).fold{...}{_ => }
```
in the code